dms/models/directory.py

```diff
@@ -622,15 +618,23 @@ def _process_message(self, msg_dict, extra_values=False):
     @api.model_create_multi
     def create(self, vals_list):
         for vals in vals_list:
-            if vals.get("root_storage_id", False):
-                vals["storage_id"] = vals["root_storage_id"]
             if vals.get("parent_id", False):
                 parent = self.browse([vals["parent_id"]])
                 data = next(iter(parent.sudo().read(["storage_id"])), {})
                 vals["storage_id"] = self._convert_to_write(data).get("storage_id")
         return super().create(vals_list)
 
     def write(self, vals):
+        if "storage_id" in vals and vals["storage_id"]:
```

Comment:
Simplify expressions whenever you are able to:
```suggestion
        if vals.get("storage_id"):
```